checker/src/org/checkerframework/checker/lowerbound/LowerBoundAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,795 @@
+        }
+
+        /** Call increment and decrement helper functions. */
+        @Override
+        public Void visitUnary(UnaryTree tree, AnnotatedTypeMirror typeDst) {
+            AnnotatedTypeMirror typeSrc = getAnnotatedType(tree.getExpression());
+            switch (tree.getKind()) {
+                case PREFIX_INCREMENT:
+                    promoteType(typeSrc, typeDst);
+                    break;
+                case PREFIX_DECREMENT:
+                    demoteType(typeSrc, typeDst);
+                    break;
+                case POSTFIX_INCREMENT:
+                case POSTFIX_DECREMENT:
+                    // Do nothing. The CF should take care of these itself.
+                    break;
+                default:
+                    break;
+            }
+            return super.visitUnary(tree, typeDst);
+        }
+
+        /** Special handling for min and max from Math. Min is LUB, max is GLB. */
+        @Override
+        public Void visitMethodInvocation(MethodInvocationTree tree, AnnotatedTypeMirror type) {
+
+            ExecutableElement min = TreeUtils.getMethod("java.lang.Math", "min", 2, processingEnv);
+            ExecutableElement max = TreeUtils.getMethod("java.lang.Math", "max", 2, processingEnv);
+
+            if (TreeUtils.isMethodInvocation(tree, min, processingEnv)) {
+                ExpressionTree left = tree.getArguments().get(0);
+                ExpressionTree right = tree.getArguments().get(1);
+                type.replaceAnnotation(
+                        qualHierarchy.leastUpperBound(
+                                getAnnotatedType(left).getAnnotationInHierarchy(POS),
+                                getAnnotatedType(right).getAnnotationInHierarchy(POS)));
+            }
+            if (TreeUtils.isMethodInvocation(tree, max, processingEnv)) {
+                ExpressionTree left = tree.getArguments().get(0);
+                ExpressionTree right = tree.getArguments().get(1);
+                type.replaceAnnotation(
+                        qualHierarchy.greatestLowerBound(
+                                getAnnotatedType(left).getAnnotationInHierarchy(POS),
+                                getAnnotatedType(right).getAnnotationInHierarchy(POS)));
+            }
+            return super.visitMethodInvocation(tree, type);
+        }
+
+        /**
+         * If the argument valueType indicates that the Constant Value Checker knows the exact value
+         * of the annotated expression, returns that integer. Otherwise returns null. This method
+         * should only be used by clients who need exactly one value - such as the binary operator
+         * rules - and not by those that need to know whether a valueType belongs to an LBC
+         * qualifier. Clients needing a qualifier should use lowerBoundAnmFromValueType instead of
+         * this method.
+         */
+        private Integer maybeValFromValueType(AnnotatedTypeMirror valueType) {
```

Comment:
I used the "maybe" name in reference to the [Maybe](https://wiki.haskell.org/Maybe) type in haskell. I understand that not being a good name in Java, though, and I'll make the suggested changes.